tmt/steps/execute/__init__.py

```diff
@@ -243,6 +243,82 @@ def reboot_requested(self) -> bool:
         """ Whether a guest reboot has been requested while the test was running """
         return self.soft_reboot_requested or self.hard_reboot_requested
 
+    @property
+    def restart_requested(self) -> bool:
+        """ Whether a test restart has been requested """
+
+        return self.return_code in self.test.restart_test_on_exit_code
+
+    @property
+    def is_guest_alive(self) -> bool:
```

Comment:
I'm afraid that this name is deceiving - without looking for the note I'd assume it checks machine is up/running.

Naming is really hard, but imo we need another name (currently it just signals that restart/reboot is requested)

Reply:
Might be. I would like to avoid any additional checks, it's all based on assumptions: when the watchdog realizes the guest is not responding, or when the test crashes, the guest is most likely in a sensitive situation, and I believe it would be very error-prone to try to check how it's doing, in any way. So tmt assumes a defensive posture if any of these happens: the guest is probably better off being left alone, don't try to push/pull, don't try to run AVC check on it, etc. It's like a barrier that's put up by these "sensors", and until it's removed by a hard reboot or (successful?) re-run of the test, it would be risky to treat the guest as if it could do everything a healthy guest can.

So, now that you know the context, feel free to propose a name :) Or even a different approach

Reply:
@lukaszachy I guess we are still waiting for some better naming proposals?

Reply:
Ok, here's some brainstorm from me:

* `is_guest_ok()`
* `is_guest_healthy()`
* `any_guest_doubts()`
* `guest_seems_to_be_ok()`

Reply:
+1 to `is_guest_ok` and `is_guest_healthy`.

Reply:
Renamed to `is_guest_healthy` in 24de2407

Reply:
Ack, thanks.